kernel/Kconfig

```diff
@@ -436,6 +436,113 @@ menuconfig OBJ_CORE
 	  automated means.
 
 if OBJ_CORE
+config OBJ_CORE_CONDVAR
+	bool "Integrate condition variables into object core framework"
+	default y
+	help
+	  When enabled, this option integrates condition variables into the
+	  object core framework.
+
+config OBJ_CORE_EVENT
+	bool "Integrate events into object core framework"
+	default y if EVENTS
+	help
+	  When enabled, this option integrate kernel events into the object
+	  core framework.
+
+config OBJ_CORE_FIFO
+	bool "Integrate FIFOs into object core framework"
```

Comment:
Seems odd to have fifos/lifos share all the same implementation except... tracking and stats?  Shouldn't you just add the obj_core to k_queue instead?

Reply:
That was the original plan until I realised that interferes with the K_OBJ_CORE() macro. That is, the caller to K_OBJ_CORE() when using fifo/lifo's would have to pass a pointer to the underlying k_queue (somewhat exposing implementation details), or we would need a K_FIFO_OBJ_CORE() / K_LIFO_OBJ_CORE(). I wasn't particularly fond of either of those options.